aws/runner.go

```diff
@@ -10,31 +13,58 @@ import (
 type Runner struct {
 	tflint.Runner
 	PluginConfig *Config
-	AwsClient    *Client
+	AwsClients   map[string]*Client
 }
 
 // NewRunner returns a custom AWS runner.
 func NewRunner(runner tflint.Runner, config *Config) (*Runner, error) {
-	var client *Client
+	clients := map[string]*Client{
+		"aws": nil,
+	}
 	if config.DeepCheck {
 		credentials, err := GetCredentialsFromProvider(runner)
 		if err != nil {
 			return nil, err
 		}
-
-		client, err = NewClient(config.toCredentials().Merge(credentials))
-		if err != nil {
-			return nil, err
+		for k, cred := range credentials {
+			client, err := NewClient(config.toCredentials().Merge(cred))
+			if err != nil {
+				return nil, err
+			}
+			clients[k] = client
```

Comment:
I noticed that the client is not initialized as before if you don't declare any provider blocks. Previously, if you wrote a credential in `.tflint.hcl`, you should have been able to enable deep checking without a provider block. e.g.

```hcl
plugin "aws" {
  enabled = true
  access_key = "AWS_ACCESS_KEY",
  secret_key = "AWS_SECRET_KEY",
}
```

```hcl
resource "aws_instance" "foo" {
  ami = "invalid"
}
```

Reply:
f73855a2e4d5ab667d848e10eaa0c559f47febfa

Reply:
https://github.com/terraform-linters/tflint-ruleset-aws/blob/master/docs/deep_checking.md#shared-credentials

> If these configurations are defined in the provider block, they will also be taken into account. But the priority is lower than the above way.

Hmm... Maybe we have to fix f73855a according to the priority.

Reply:
The priority of `plugin` setting is higher than `provider`.

Reply:
Fixed. 313f0da9624d402e23056bdba9b990fe202e1cd8